### PR TITLE
Index date

### PR DESCRIPTION
**Pull request by LFISHER7**
Updates the analysis code to use the end of the month as the comparison date for event 2 (see https://github.com/opensafely-core/job-server/issues/2694).

Makes a first pass at adding more explanatory content to the report, addressing https://github.com/opensafely-core/interactive-templates/issues/10

### CODE REVIEW

File: templates/v2/analysis/render_report.py [hunk 1]
```diff
@@ -94,27 +94,31 @@ def get_data(
...
         "sex": {
             "title": "Sex",
-            "description": "Sex breakdown",
+            "description": "",
             "figure": figure_paths["sex"],
         },
         "imd": {
             "title": "Index of Multiple Deprivation",
-            "description": "Index of Multiple Deprivation breakdown",
+            "description": "Index of Multiple Deprivation breakdown is presented as quintiles, based on English indices of deprivation 2019. These quintile range from 1 (most deprived) to 5 (least deprived See the link below for more details.",
```
Comment (ghickman) [inline templates/v2/analysis/render_report.py:114]: Missing a `)` after `least deprived`.

File: templates/v2/analysis/report_template.html [hunk 2]
```diff
@@ -1,43 +1,49 @@
...
+                        code recorded in their primary care record from <a
+                            href="{{ codelist_1_link }}"> this <strong>{{
+                                codelist_1_name }}</strong>
+                            codelist</a> each month in the period between {{
+                        start_date }} and {{ end_date }} who also had a code
+                        recorded from <a href="{{ codelist_2_link }}"> this
+                            codelist for
+                            <strong>{{ codelist_2_name }}</strong></a> in the
+                        same month or up to <strong>{{ time_value }} {{
+                            time_scale }} {{ time_event }}</strong> the last day
```
Comment (ghickman) [inline templates/v2/analysis/report_template.html:24]: Have you tested that the templating still works with the variable interpolations split over newlines like this?

**Review by ghickman — APPROVED** [review]
Approved but there's a couple of bits for you to look at 🙂

File: templates/v2/analysis/report_template.html (hunk 2 above)
Comment (lucyb) [reply templates/v2/analysis/report_template.html:24]: I was curious, so I tested it locally using smush.py and the templating does seem to work okay.

Comment (LFISHER7) [reply templates/v2/analysis/report_template.html:24]: It all looks ok when run locally

**Review by lucyb — APPROVED** [review]

**Pull request merged**